scipy/integrate/_cubature.py

```diff
@@ -0,0 +1,416 @@
+import math
+import heapq
+import functools
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+import numpy as np
+
+
+class CubatureRule(ABC):
+    def __init__(self, nodes, weights):
+        self.nodes = nodes
+        self.weights = weights
+
+    def dimension(self):
```

Comment:
Think we just want to call this `ndim` for consistency.

Reply:
... and make it a property